### ISSUE TITLE
[BUG] MTP inference confuse

### ISSUE THREAD

**Issue by niyunsheng** [deepseek-ai/DeepSeek-V3#712]
**Describe the bug**

In the paper, hnrom is before enorm, but in the open source code of sglang, enorm is before hnorm.

I do think the code in SGLang is Right, `torch.cat((enorm_hidden, hnorm_hidden),dim=-1)`.

* paper
![Image](https://github.com/user-attachments/assets/408f50cf-4311-4283-8554-1b36ff9944bd)

* [SGLang inference code](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/models/deepseek_nextn.py#L88)

![Image](https://github.com/user-attachments/assets/c3354585-af00-4564-8ed7-b11d244d843d)
